PatternPal/PatternPal/Services/StepByStepService.cs

```diff
@@ -1,41 +1,211 @@
-﻿using PatternPal.Core.StepByStep;
+﻿#region
+
+using System.Reflection;
+
+using PatternPal.Core.Recognizers;
+using PatternPal.Core.StepByStep;
 using PatternPal.SyntaxTree;
 using PatternPal.SyntaxTree.Abstractions.Entities;
 
+using InstructionSet = PatternPal.Core.StepByStep.InstructionSet;
+
+#endregion
+
 namespace PatternPal.Services;
 
 public class StepByStepService : Protos.StepByStepService.StepByStepServiceBase
 {
+    /// <inheritdoc />
     public override Task< GetInstructionSetsResponse > GetInstructionSets(
         GetInstructionSetsRequest request,
         ServerCallContext context)
     {
-        throw new NotImplementedException();
+        GetInstructionSetsResponse response = new();
+
+        // Obtain all the enum values in the proto.
+        bool initialValue = true;
+        foreach (Recognizer test in Enum.GetValues< Recognizer >())
+        {
+            if (initialValue)
+            {
+                initialValue = false;
+                continue;
+            }
+            response.Recognizers.Add(test);
+        }
+
+        return Task.FromResult(response);
     }
 
+    /// <inheritdoc />
+    public override Task< GetInstructionSetResponse > GetInstructionSet(
+        GetInstructionSetRequest request,
+        ServerCallContext context)
+    {
+        IRecognizer recognizer = GetRecognizer(request.SelectedCombobox);
+        List< IInstruction > instructions = recognizer.GenerateStepsList();
+
+        InstructionSetHolder.instructionSet =
+            new InstructionSet(
+                recognizer,
+                instructions);
+
+        Protos.InstructionSet res =
+            new()
+            {
+                Name = recognizer.Name,
+                NumberOfInstructions = (uint)instructions.Count
+            };
+
+        GetInstructionSetResponse response = new()
+                                             {
+                                                 SelectedInstructionset = res
+                                             };
+
+        return Task.FromResult(response);
+    }
+
+    /// <inheritdoc />
     public override Task< GetSelectableClassesResponse > GetSelectableClasses(
         GetSelectableClassesRequest request,
         ServerCallContext context)
     {
-        throw new NotImplementedException();
+        SyntaxGraph graph = new();
```

Comment:
See comment regarding this in the PR.

Reply:
This was removed and made more intuitive with a OpenFileDialog where the user can provide which files they want recognized and in the "new" option the user always provides a save location for the new file.